### PR TITLE
Ported Decimation example from Cxx to Python

### PR DESCRIPTION
**Pull request by bdemin**

### CODE REVIEW

File: src/Python/Meshes/Decimation.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+def get_program_parameters():
+    import argparse
+    description = 'Decimate polydata.'
+    epilogue = '''
+    This is an example using vtkDecimatePro to decimate input polydata, if provided, or a sphere otherwise.
+    '''
+    parser = argparse.ArgumentParser(description=description, epilog=epilogue)
+    parser.add_argument('filename', nargs='?', default=None, help='Optional input filename e.g cow.g.')
+    parser.add_argument('reduction', nargs='?', type = float, default=None, help='Optional reduction percentage e.g 0.9')
+    args = parser.parse_args()
```
Comment (ajpmaclean) [inline src/Python/Meshes/Decimation.py]: Set the default to `0.9` and change the help text to: `Sets the decimation target reduction, (default is 0.9)`.

File: src/Python/Meshes/Decimation.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    if filePath and os.path.isfile(filePath):
+        inputPolyData = ReadPolyData(filePath)
+        if not inputPolyData:
+            inputPolyData = GetSpherePD()
+    else:
+        inputPolyData = GetSpherePD()
+
+    if not reduction:
+        reduction = .9 # 90% reduction
+
```
Comment (ajpmaclean) [inline src/Python/Meshes/Decimation.py]: The above two lines are not needed if the default is set to 0.9 in `get_program_parameters()`.

File: src/Python/Meshes/Decimation.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    # (xmin, ymin, xmax, ymax)
+    leftViewport = [0.0, 0.0, 0.5, 1.0]
+    rightViewport = [0.5, 0.0, 1.0, 1.0]
+
+    # Setup both renderers
+    leftRenderer = vtk.vtkRenderer()
+    renderWindow.AddRenderer(leftRenderer)
+    leftRenderer.SetViewport(leftViewport)
+    leftRenderer.SetBackground(.6, .5, .4)
+
```
Comment (ajpmaclean) [inline src/Python/Meshes/Decimation.py]: Use `vtkNamedColors` and you can do something like this:
``` Python
    colors.SetColor('leftBkg', [0.6, 0.5, 0.4, 1.0])
    colors.SetColor('rightBkg', [0.4, 0.5, 0.6, 1.0])
```
Here you have added some new named colors the colors that you can subsequently use.
Put these lines up where you define the colors, after line 28.
Then you can use `leftRenderer.SetBackground(colors.GetColor3d('leftBkg'))`

File: src/Python/Meshes/Decimation.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    rightRenderer = vtk.vtkRenderer()
+    renderWindow.AddRenderer(rightRenderer)
+    rightRenderer.SetViewport(rightViewport)
+    rightRenderer.SetBackground(.4, .5, .6)
+
```
Comment (ajpmaclean) [inline src/Python/Meshes/Decimation.py]: Replace this line with: `rightRenderer.SetBackground(colors.GetColor3d('rightBkg'))`

File: src/Python/Meshes/Decimation.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+def get_program_parameters():
+    import argparse
+    description = 'Decimate polydata.'
+    epilogue = '''
+    This is an example using vtkDecimatePro to decimate input polydata, if provided, or a sphere otherwise.
+    '''
+    parser = argparse.ArgumentParser(description=description, epilog=epilogue)
+    parser.add_argument('filename', nargs='?', default=None, help='Optional input filename e.g cow.g.')
+    parser.add_argument('reduction', nargs='?', type = float, default=None, help='Optional reduction percentage e.g 0.9')
```
Comment (ajpmaclean) [inline src/Python/Meshes/Decimation.py]: Change `cow.g` to `Torso.vtp`.

**Comment by ajpmaclean** [comment]
Looks good, subject to the above changes. I really like the use of `def ReadPolyData(file_name)`!

**Comment by bdemin** [comment]
Thanks for the comments. All the relevant changes have been added.

**Comment by ajpmaclean** [comment]
Thankyou for your contribution.

**Pull request merged**